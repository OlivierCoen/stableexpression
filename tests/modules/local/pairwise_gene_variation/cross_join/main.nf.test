nextflow_process {

    name "Test Process CROSS_JOIN"
    script "modules/local/pairwise_gene_variation/cross_join/main.nf"
    process "CROSS_JOIN"
    tag "cross_join"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                file_1 = file( '$projectDir/tests/data/pairwise_gene_variation/cross_join/input/count_chunk.0.parquet', checkIfExists: true)
                file_2 = file( '$projectDir/tests/data/pairwise_gene_variation/cross_join/input/count_chunk.1.parquet', checkIfExists: true)
                input[0] = [file_1, file_2]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
