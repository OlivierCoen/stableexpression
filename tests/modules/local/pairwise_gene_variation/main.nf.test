nextflow_process {

    name "Test Process PAIRWISE_GENE_VARIATION"
    script "modules/local/pairwise_gene_variation/main.nf"
    process "PAIRWISE_GENE_VARIATION"
    tag "pairwisegenevariation"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/data/merge_counts/output/all_counts.parquet', checkIfExists: true)
                input[0] = ch_counts
                """
            }
        }

        then {
            println(process.out)
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
