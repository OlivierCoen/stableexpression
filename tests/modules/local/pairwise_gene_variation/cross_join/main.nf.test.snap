{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "cross_join.parquet:md5,5a504aab5ae5234e50b324581a0d9b7a"
                ],
                "1": [
                    [
                        "CROSS_JOIN",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "CROSS_JOIN",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "data": [
                    "cross_join.parquet:md5,5a504aab5ae5234e50b324581a0d9b7a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T10:49:03.746839728"
    }
}